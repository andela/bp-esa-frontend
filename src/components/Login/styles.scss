/* Styling the Login component */
.login-page {
   background-image: url('../../assets/dotted-worldmap.png');
   background-color: rgba(51,89,223,0.95);
   height: -webkit-fill-available;
   background-size: cover;
   background-repeat: no-repeat;
}

.login-container {
    height: fit-content;
    width: fit-content;
    margin-left: 200px;
    padding-top: 12%;
}

.top-blue {
    height: 300px;
    width: 350px;
    display: inline-block;
    border-radius: 5px 5px 0 0;
    background-color: rgba(51,89,223,0.95);
    box-shadow: 0 10px 30px 0 rgba(0,0,0,0.3);
    padding-top: 45px;
    position: relative;
}

.bottom-white {
    height: 200px;
    width: 350px;
    border-radius: 0 0 5px 5px;
    background-color: #FFF;
    box-shadow: 0 10px 30px 0 rgba(0,0,0,0.3);
    display: flex;
    align-items: center;
    justify-content: center;
}

.andela-logo {
    height: 70px;
    width: 70px;
    margin: auto;

}

.a {
    height: 65px;
    width: 65px;
    color: #FFF;
    font-family: Arial;
    font-size: 36px;
    line-height: 40px;
    position: absolute;
    left: 46%;
    top: 19%;
}

.logo {
    width: 65px;
    height: 65px;
    position: absolute;
}

.engagement-support-automation {
    height: 20px;
    color: #FFF;
    font-family: Arial;
    text-align: center;
    font-size: 18px;
    line-height: 20px;
    margin-top: 25px;
}

.sign-in-to-get-started {
    height: 20px;
    color: #FFF;
    font-family: Arial;
    font-size: 18px;
    font-style: italic;
    text-align: center;
    line-height: 20px;
    margin-top: 25px;
}

.sign-in-button-container {
    box-sizing: border-box;
    height: 48px;
    width: 254px;
    border-radius: 5px;
    background-color: #FFF;
    border: 1px solid #CCD;
    display: flex;
    align-items: center;
}

.google-logo {
    height: 28px;
    width: 27.34px;
  }

.sign-in-with-google {
    height: 20px;
    width: 156px;
    color: #667;
    font-family: Arial;
    font-size: 18px;
    line-height: 20px;
    margin-left: 8px;
}

@media screen and (max-width: 800px){
    .login-container{
        margin: auto;
        padding-top: 30%;
    }
}

@media screen and (max-width: 1100px){
    .login-container{
        margin: auto;
        padding-top: 35%;
    }
}

@media screen and (max-width: 1100px) and (max-height: 1000px){
    .login-container{
        margin: auto;
        padding-top: 15%;
    }
}

@media screen and (max-width: 800px) and (max-height: 600px){
    .login-container{
        margin: auto;
        padding-top: 5%;
    }
    .top-blue {
        height: 210px;
        width: 250px;
        padding-top: 15%;
    }
    .a {
        left: 44.5%;
        top: 24%;
    }
    .engagement-support-automation {
        font-size: 14px;
        margin-top: 15px;
    }
    .sign-in-to-get-started {
        font-size: 14px;
        margin-top: 15px;
    }
    .bottom-white {
        height: 120px;
        width: 250px;
    }
    .sign-in-button-container {
        height: 48px;
        width: 170px;
    }    
   .google-logo {
        height: 24px;
        width: 23px;
    }    
    .sign-in-with-google {
        font-size: 14px;
        margin-left: 3px;
    }
}

@media screen and (min-width: 800px) and (max-height: 600px){
    .login-container{
        margin: auto;
        padding-top: 5%;
    }
    .top-blue {
        height: 210px;
        width: 250px;
        padding-top: 15%;
        }
    .a{
        left: 44.5%;
        top: 24%;
    }
    .engagement-support-automation {
        font-size: 14px;
        margin-top: 15px;
    }
    .sign-in-to-get-started {
        font-size: 14px;
        margin-top: 15px;
    }
    .bottom-white {
        height: 120px;
        width: 250px;
    }
    .sign-in-button-container {
        height: 48px;
        width: 170px;
    }
    .google-logo {
        height: 24px;
        width: 23px;
    }    
    .sign-in-with-google {
        font-size: 14px;
        margin-left: 3px;
    }
}



@media screen and (max-width: 600px) and (max-height: 600px){
   .login-container{
       margin: auto;
       padding-top: 25%;
    }
   .top-blue {
        height: 250px;
        width: 250px;
        padding-top: 15%;
    }
    .a{
        left: 44%;
        top: 19%;
    }
   .engagement-support-automation {
        font-size: 14px;
        margin-top: 15px;
    }
    .sign-in-to-get-started {
        font-size: 14px;
        margin-top: 15px;
    }

    .bottom-white {
        height: 150px;
        width: 250px;
    }
    .sign-in-button-container {
        height: 48px;
        width: 170px;
    }

    .google-logo {
        height: 24px;
        width: 23px;
    }

    .sign-in-with-google {
        font-size: 14px;
        margin-left: 3px;
    }
}

@media screen and (max-width: 600px) and (max-height: 700px){
    .login-container{
        margin: auto;
        padding-top: 30%;
     }
    .top-blue {
        height: 250px;
        width: 250px;
        padding-top: 15%;
    }
    .a{
        left: 44.5%;
        top: 19%;
    }
    .engagement-support-automation {
        font-size: 14px;
        margin-top: 15px;
    }
    
    .sign-in-to-get-started {
        font-size: 14px;
        margin-top: 15px;
    }
    
    .bottom-white {
        height: 150px;
        width: 250px;
    }
    .sign-in-button-container {
        height: 48px;
        width: 170px;
    }
    
    .google-logo {
        height: 24px;
        width: 23px;
    }
    
    .sign-in-with-google {
        font-size: 14px;
        margin-left: 3px;
    }
 }

 @media screen and (max-width: 600px) and (max-height: 800px) and (min-height: 700px){
    .login-container{
        margin: auto;
        padding-top: 30%;
     }
    .top-blue {
        height: 250px;
        width: 250px;
        padding-top: 15%;
    }
    .a {
        left: 44%;
        top: 19%;
    }
    .engagement-support-automation {
        font-size: 14px;
        margin-top: 15px;
    }
    .sign-in-to-get-started {
        font-size: 14px;
        margin-top: 15px;
    }
    
    .bottom-white {
        height: 150px;
        width: 250px;
    }
    .sign-in-button-container {
        height: 48px;
        width: 170px;
    }
    .google-logo {
        height: 24px;
        width: 23px;
    }
    .sign-in-with-google {
        font-size: 14px;
        margin-left: 3px;
    }
}

@media screen and (max-width: 600px) and (max-height: 900px) and (min-height: 800px){
    .login-container{
        margin: auto;
        padding-top: 45%;
    }
    .top-blue {
        height: 250px;
        width: 250px;
        padding-top: 15%;
    }
    .a {
        left: 44.5%;
        top: 20%;
    }
    .engagement-support-automation {
        font-size: 14px;
        margin-top: 15px;
    }
    
    .sign-in-to-get-started {
        font-size: 14px;
        margin-top: 15px;
    }
    
    .bottom-white {
        height: 150px;
        width: 250px;
    }
    .sign-in-button-container {
        height: 48px;
        width: 170px;
    }
    
    .google-logo {
        height: 24px;
        width: 23px;
    }
    
    .sign-in-with-google {
        font-size: 14px;
        margin-left: 3px;
    }
}
