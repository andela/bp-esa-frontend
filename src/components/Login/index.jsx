import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { notify } from 'react-notify-toast';
import { doSignInWithGoogle } from '../../firebase';
import './styles.scss';

class Login extends Component {
  onLogin = () => {
    const { setCurrentUser, history } = this.props;
    doSignInWithGoogle().then((user) => {
      if (user.user.email.match(/.*@andela.com$/)) {
        setCurrentUser(user);
        history.push('/');
        notify.show(`Hello ${user.user.displayName} ! You have Logged in Successfully!`, 'success');
      } else {
        notify.show('Login failed! Please login with your andela email!', 'error');
      }
    })
      .catch((error) => {
        if (error) {
          notify.show('Unable to login with google!', 'error');
        }
      });
  }

  render() {
    return (
      <div className="login-page">
        <div className="login-container">
          <div className="top-blue">
            <div className="andela-logo">
              <h1 className="a">A</h1>
              <img className="logo" src="/logo.png" alt="Andela Logo" />
            </div>
            <h1 className="engagement-support-automation">Engagement Support Automation</h1>
            <h1 className="sign-in-to-get-started">Sign in to get started</h1>
          </div>
          <div className="bottom-white">
            <button type="button" className="sign-in-button-container" onClick={() => this.onLogin()}>
              <img className="google-logo" src="/google.png" alt="google-icon" />
              <span className="sign-in-with-google">Sign in with Google</span>
            </button>
          </div>
        </div>
      </div>
    );
  }
}

Login.propTypes = {
  setCurrentUser: PropTypes.func,
  history: PropTypes.object.isRequired,
};

Login.defaultProps = {
  setCurrentUser: () => {},
};

export default Login;
