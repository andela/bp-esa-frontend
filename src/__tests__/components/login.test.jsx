import React from 'react';
import { shallow } from 'enzyme';
import { notify } from 'react-notify-toast';
import Login from '../../components/login';

jest.mock('../../firebase');

const props = {
  history: {
    push: jest.fn(),
  },
  setCurrentUser: jest.fn(),
};
let wrapper;

const getComponent = () => {
  if (!wrapper) {
    wrapper = shallow(<Login {...props} />);
  }
  return wrapper;
};

it('renders without crashing', () => {
  const component = getComponent();
  expect(component).toMatchSnapshot();
});

describe('onLogin() method', () => {
  Object.defineProperty(notify, 'show', { value: () => jest.fn(), writable: true });
  it('should call onLogin()', () => {
    const renderedComponent = getComponent();
    const spy = jest.spyOn(renderedComponent.instance(), 'onLogin');
    renderedComponent.find('a').simulate('click');
    expect(spy).toHaveBeenCalled();
  });

  it('should redirect to the Andela API', () => {
    const redirectUrl = `https://api-prod.andela.com/login?redirect_url=${process.env.REACT_APP_URL}`;
    window.location.replace = jest.fn();
    wrapper.find('.loginUrl').simulate('click');
    expect(window.location.replace).toHaveBeenCalledWith(redirectUrl);
  });
});
