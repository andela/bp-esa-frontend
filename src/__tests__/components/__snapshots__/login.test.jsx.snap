// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Login
    history={
      Object {
        "push": [MockFunction],
      }
    }
    setCurrentUser={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="login_details"
      >
        <div
          className="login_logo"
        >
          <div>
            A
          </div>
          <img
            alt="Andela Logo"
            className="logo"
            src="/logo.png"
          />
        </div>
        <div
          className="login_title"
        >
          <h1>
            ESA
          </h1>
        </div>
        <div
          className="login_link"
        >
          <a
            href="#"
            onClick={[Function]}
          >
            <img
              alt="google-icon"
              src="/google.png"
            />
            <span>
              LOGIN TO GET STARTED
            </span>
          </a>
        </div>
      </div>,
      "className": "login_page",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="login_logo"
          >
            <div>
              A
            </div>
            <img
              alt="Andela Logo"
              className="logo"
              src="/logo.png"
            />
          </div>,
          <div
            className="login_title"
          >
            <h1>
              ESA
            </h1>
          </div>,
          <div
            className="login_link"
          >
            <a
              href="#"
              onClick={[Function]}
            >
              <img
                alt="google-icon"
                src="/google.png"
              />
              <span>
                LOGIN TO GET STARTED
              </span>
            </a>
          </div>,
        ],
        "className": "login_details",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div>
                A
              </div>,
              <img
                alt="Andela Logo"
                className="logo"
                src="/logo.png"
              />,
            ],
            "className": "login_logo",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "A",
              },
              "ref": null,
              "rendered": "A",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "Andela Logo",
                "className": "logo",
                "src": "/logo.png",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h1>
              ESA
            </h1>,
            "className": "login_title",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "ESA",
            },
            "ref": null,
            "rendered": "ESA",
            "type": "h1",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <a
              href="#"
              onClick={[Function]}
            >
              <img
                alt="google-icon"
                src="/google.png"
              />
              <span>
                LOGIN TO GET STARTED
              </span>
            </a>,
            "className": "login_link",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <img
                  alt="google-icon"
                  src="/google.png"
                />,
                <span>
                  LOGIN TO GET STARTED
                </span>,
              ],
              "href": "#",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "google-icon",
                  "src": "/google.png",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "LOGIN TO GET STARTED",
                },
                "ref": null,
                "rendered": "LOGIN TO GET STARTED",
                "type": "span",
              },
            ],
            "type": "a",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="login_details"
        >
          <div
            className="login_logo"
          >
            <div>
              A
            </div>
            <img
              alt="Andela Logo"
              className="logo"
              src="/logo.png"
            />
          </div>
          <div
            className="login_title"
          >
            <h1>
              ESA
            </h1>
          </div>
          <div
            className="login_link"
          >
            <a
              href="#"
              onClick={[Function]}
            >
              <img
                alt="google-icon"
                src="/google.png"
              />
              <span>
                LOGIN TO GET STARTED
              </span>
            </a>
          </div>
        </div>,
        "className": "login_page",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="login_logo"
            >
              <div>
                A
              </div>
              <img
                alt="Andela Logo"
                className="logo"
                src="/logo.png"
              />
            </div>,
            <div
              className="login_title"
            >
              <h1>
                ESA
              </h1>
            </div>,
            <div
              className="login_link"
            >
              <a
                href="#"
                onClick={[Function]}
              >
                <img
                  alt="google-icon"
                  src="/google.png"
                />
                <span>
                  LOGIN TO GET STARTED
                </span>
              </a>
            </div>,
          ],
          "className": "login_details",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div>
                  A
                </div>,
                <img
                  alt="Andela Logo"
                  className="logo"
                  src="/logo.png"
                />,
              ],
              "className": "login_logo",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "A",
                },
                "ref": null,
                "rendered": "A",
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "Andela Logo",
                  "className": "logo",
                  "src": "/logo.png",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <h1>
                ESA
              </h1>,
              "className": "login_title",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "ESA",
              },
              "ref": null,
              "rendered": "ESA",
              "type": "h1",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <a
                href="#"
                onClick={[Function]}
              >
                <img
                  alt="google-icon"
                  src="/google.png"
                />
                <span>
                  LOGIN TO GET STARTED
                </span>
              </a>,
              "className": "login_link",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <img
                    alt="google-icon"
                    src="/google.png"
                  />,
                  <span>
                    LOGIN TO GET STARTED
                  </span>,
                ],
                "href": "#",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "alt": "google-icon",
                    "src": "/google.png",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "LOGIN TO GET STARTED",
                  },
                  "ref": null,
                  "rendered": "LOGIN TO GET STARTED",
                  "type": "span",
                },
              ],
              "type": "a",
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
